worker/log.go

```diff
@@ -0,0 +1,141 @@
+package worker
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"github.com/fsnotify/fsnotify"
+	"github.com/sirupsen/logrus"
+	"io"
+	"os"
+	"path/filepath"
+	"sync"
+)
+
+//The buffer size 1024 is just chosen randomly, performance is only affected when the process writes huge amount of data to the file
+const BUFFER_SIZE = 1024
```

Comment:
nit: The `SCREAMING_SNAKE_CASE` convention for constants isn't popular in Go.

```suggestion
const BufferSize = 1024
```